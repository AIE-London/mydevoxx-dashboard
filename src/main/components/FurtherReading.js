/**
 * Created by tsadler on 07/04/2017.
 */
import React from 'react';
import {Grid, Row, Col} from 'react-flexbox-grid';
import styled from 'styled-components';
import Card from './Card';


const StatMainHeader = styled.h1`
    text-align: left;
    font-size: 2.5em;
    font-family: helvetica;
`;

const StatValue = styled.p`
    text-align: center;
    font-size: 1.5em;
    font-family: helvetica;
`;

const LinkValue = styled.p`
    text-align: center;
    font-size: 1em;
    color: #fff;
    font-family: helvetica;
`;

class furtherReading extends React.Component {
    render() {

        return <Card>
            <Row center="xs">
                    <Col xs={12}>


                        <Col xs={2}>
                            <Row start="xs">
                                <StatMainHeader>AdvancedReading</StatMainHeader>
                                <LinkValue> <a href> http:www.bbc.com </a> </LinkValue>
                            </Row>
                        </Col>

                        <Row center="xs">
                            <Col xs={10}>

                                <StatValue>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur at gravida neque. Suspendisse
                                    vitae lacus eget tellus facilisis finibus. Integer vestibulum, risus vel molestie ornare,
                                    urna lacus posuere libero, in convallis ipsum enim at eros.... </StatValue>

                            </Col>
                        </Row>
                    </Col>

                </Row>
            </Card>

    }
}

export default furtherReading;