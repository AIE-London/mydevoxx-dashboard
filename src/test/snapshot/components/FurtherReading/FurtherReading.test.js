/**
 * Created by tsadler on 07/04/2017.
 */
import React from "react";
import ReactDOM from "react-dom";
import FurtherReading from "../../../../main/components/FurtherReading";
import renderer from "react-test-renderer";

test("FurtherReading component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<FutherReading title="Further Reading" />, div);
  });
});

test("FurtherReading component with title snapshot", () => {
  let data = {
    url: "http://www.bbc.co.uk"
  };

  const tree = renderer
    .create(
      <FurtherReading
        mainHeader="Advanced Reading"
        url="http://www.bbc.co.uk"
        urlText="Link to advanced reading"
        text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur at gravida neque.Lacus eget tellus facilisis...."
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
