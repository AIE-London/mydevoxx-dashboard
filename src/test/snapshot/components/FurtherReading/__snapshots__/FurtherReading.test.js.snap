exports[`test FurtherReading component with title snapshot 1`] = `
<section
  className="eieVzi"
  id="recommended-reading">
  <h2
    className="fcdRkU">
    Recommendations
  </h2>
  <div>
    <a
      href="http://devoxx.co.uk">
      <div
        className="hRXQwI row">
        <div>
          <div>
            <h3>
              Intro to Devoxx
            </h3>
            <h4>
              Devoxx Webpage
            </h4>
          </div>
        </div>
        <p>
          Based on your 
          tracks
        </p>
      </div>
    </a>
    <a
      href="http://devoxx.co.uk">
      <div
        className="hRXQwI row">
        <div>
          <div>
            <h3>
              Intro to Devoxx2
            </h3>
            <h4>
              Devoxx Webpage
            </h4>
          </div>
        </div>
        <p>
          Based on your 
          tracks
        </p>
      </div>
    </a>
    <a
      href="http://devoxx.co.uk">
      <div
        className="hRXQwI row">
        <div>
          <div>
            <h3>
              Intro to Devoxx3
            </h3>
            <h4>
              Devoxx Webpage
            </h4>
          </div>
        </div>
        <p>
          Based on your 
          tracks
        </p>
      </div>
    </a>
    <a
      href="http://devoxx.co.uk">
      <div
        className="hRXQwI row">
        <div>
          <div>
            <h3>
              Intro to Devoxx4
            </h3>
            <h4>
              Devoxx Webpage
            </h4>
          </div>
        </div>
        <p>
          Based on your 
          tracks
        </p>
      </div>
    </a>
  </div>
</section>
`;
